app/models/category.rb

```diff
@@ -0,0 +1,12 @@
+class Category < ApplicationRecord
+  belongs_to :user
+  has_many :subcategories, class_name: 'Category', foreign_key: 'parent_id', dependent: :destroy
+  belongs_to :parent_category, class_name: 'Category', optional: true
```

Comment:
I think that you need to specify `primary_key:  :parent_id`  to make this relation works properly

Reply:
has_many :subcategories, class_name: 'Category', primary_key: :parent_id, dependent: :destroy

Reply:
understood what you mean. updated code. it was working and keeps working